nautobot/extras/models/customfields.py

```diff
@@ -325,8 +328,97 @@ def validate(self, value):
 
             # Validate selected choice
             if self.type == CustomFieldTypeChoices.TYPE_SELECT:
-                if value not in self.choices:
-                    raise ValidationError(f"Invalid choice ({value}). Available choices are: {', '.join(self.choices)}")
+                if value not in self.choices.values_list("value", flat=True):
+                    raise ValidationError(
+                        f"Invalid choice ({value}). Available choices are: {', '.join(self.choices.values_list('value', flat=True))}"
+                    )
+
+            if self.type == CustomFieldTypeChoices.TYPE_MULTISELECT:
+                if not set(value).issubset(self.choices.values_list("value", flat=True)):
+                    raise ValidationError(
+                        f"Invalid choice(s) ({value}). Available choices are: {', '.join(self.choices.values_list('value', flat=True))}"
+                    )
 
         elif self.required:
             raise ValidationError("Required field cannot be empty.")
+
+    def delete(self, *args, **kwargs):
+        """
+        Handle the cleanup of old custom field data when a CustomField is deleted.
+        """
+        content_types = set(self.content_types.values_list("pk", flat=True))
+
+        super().delete(*args, **kwargs)
+
+        delete_custom_field_data.delay(self.name, content_types)
+
+
+class CustomFieldChoice(BaseModel):
+    """
+    The custom field choice is used to store the possible set of values for a selection type custom field
+    """
+
+    field = models.ForeignKey(
+        to="extras.CustomField",
+        on_delete=models.CASCADE,
+        related_name="choices",
+        limit_choices_to={"type": CustomFieldTypeChoices.TYPE_SELECT},
```

Comment:
Doesn't this need to include TYPE_MULTISELECT as well?

Reply:
Hmm, presumably yes, however, it is working now, so I am not sure what this is actually doing then...

Reply:
So according to the docs, this only affects choices in forms and the admin. Because of the UX of the formsets in admin in this case, it doesn't actually matter. I will add it for now so it is not confusing, but we will revisit this anyway with a new UI for custom fields.